packages/fuel-indexer-graphql-dyn/src/schema/schema_type_builder.rs

```diff
@@ -0,0 +1,420 @@
+                        builder.fields.insert(
+                            field_id.clone(),
+                            DynDataFieldType {
+                                name: field.name.clone(),
+                                data_type_id: field_type_id,
+                                store_data_field_id: field_id.clone(),
+                            },
+                        );
+                    }
+                }
+                Enum(name, variants) => {
+                    let builder = self.define_enum(name.to_pascal_string());
+                    for (field_id, field) in variants {
+                        let field_type_id: DynDataTypeId = field.type_id.clone().into();
+                        builder.variants.insert(
+                            field_id.clone(),
+                            DynDataFieldType {
+                                name: field.name.clone(),
+                                data_type_id: field_type_id,
+                                store_data_field_id: field_id.clone(),
+                            },
+                        );
+                    }
+                }
+            }
+        }
+        for (id, obj) in &store.obj {
+            let (_, node) = self.define_node(id);
+            for (field_id, field) in &obj.fields {
+                let field_type_id: DynDataTypeId = field.type_id.clone().into();
+                node.define_data(field.name.clone(), field_id, field_type_id)
+            }
+        }
+        for (id, assoc) in &store.assoc {
+            let tail = assoc.tail.clone();
+            let tail: DynNodeTypeId = tail.into();
+            let head = assoc.head.clone();
+            let head: DynNodeTypeId = head.into();
+            let (_, edge) = self.define_edge(id, tail, head);
+            for (field_id, field) in &assoc.fields {
+                let field_type_id: DynDataTypeId = field.type_id.clone().into();
+                edge.define_data(field.name.clone(), field_id, field_type_id)
+            }
+        }
+        self
+    }
+    pub fn define_enum(&mut self, name: impl Into<String>) -> &mut DynEnumTypeBuilder {
+        let name = Name::new_pascal(name);
+        let id = DynDataTypeId::Name(name.clone());
+        match self
+            .data
+            .entry(id)
+            .or_insert(DynDataTypeBuilder::Enum(DynEnumTypeBuilder::new(name)))
+        {
+            DynDataTypeBuilder::Enum(builder) => builder,
+            _ => unreachable!(),
+        }
+    }
+    pub fn define_object(
+        &mut self,
+        name: impl Into<String>,
+    ) -> &mut DynObjectTypeBuilder {
+        let name = Name::new_pascal(name);
+        let id = DynDataTypeId::Name(name.clone());
+        match self
+            .data
+            .entry(id)
+            .or_insert(DynDataTypeBuilder::Object(DynObjectTypeBuilder::new(name)))
+        {
+            DynDataTypeBuilder::Object(builder) => builder,
+            _ => unreachable!(),
```

Comment:
Could we add a message to `unreachable!` that explains why this is unreachable? (In case the path gets hit) - or at least a comment